hw3/02-charts.js

```diff
@@ -1,45 +1,151 @@
+// Colors for doughnut chart sections
 const backgroundColors = [
-  'rgba(54, 162, 235, 0.8)',
-  'rgba(255, 206, 86, 0.8)',
-  'rgba(255, 99, 132, 0.8)',
-  'rgba(75, 192, 192, 0.8)',
-  'rgba(153, 102, 255, 0.8)',
-  'rgba(255, 159, 64, 0.8)',
-  'rgba(199, 199, 199, 0.8)',
-  'rgba(83, 102, 255, 0.8)',
-  'rgba(40, 159, 64, 0.8)',
-  'rgba(210, 199, 199, 0.8)',
-  'rgba(78, 52, 199, 0.8)',
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
+  "rgba(52, 162, 235, 0.8)",
+  "rgba(252, 206, 86, 0.8)",
+  "rgba(252, 99, 132, 0.8)",
+  "rgba(72, 192, 192, 0.8)",
+  "rgba(152, 102, 255, 0.8)",
+  "rgba(252, 159, 64, 0.8)",
+  "rgba(192, 199, 199, 0.8)",
+  "rgba(82, 102, 255, 0.8)",
+  "rgba(42, 159, 64, 0.8)",
+  "rgba(212, 199, 199, 0.8)",
+  "rgba(72, 52, 199, 0.8)",
 ];
 
 const borderColors = [
-  'rgba(54, 162, 235, 1)',
-  'rgba(255, 206, 86, 1)',
-  'rgba(255, 99, 132, 1)',
-  'rgba(75, 192, 192, 1)',
-  'rgba(153, 102, 255, 1)',
-  'rgba(255, 159, 64, 1)',
-  'rgba(159, 159, 159, 1)',
-  'rgba(83, 102, 255, 1)',
-  'rgba(40, 159, 64, 1)',
-  'rgba(210, 199, 199, 1)',
-  'rgba(78, 52, 199, 1)',
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
+  "rgba(52, 162, 235, 1)",
+  "rgba(252, 206, 86, 1)",
+  "rgba(252, 99, 132, 1)",
+  "rgba(72, 192, 192, 1)",
+  "rgba(152, 102, 255, 1)",
+  "rgba(252, 159, 64, 1)",
+  "rgba(152, 159, 159, 1)",
+  "rgba(82, 102, 255, 1)",
+  "rgba(42, 159, 64, 1)",
+  "rgba(212, 199, 199, 1)",
+  "rgba(72, 52, 199, 1)",
 ];
 
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters";
 
+// Collect houses as labels
+const houses = [];
+
+// Houses with their associated # of characters
+const houseCount = [];
+
+// Collect house's counts to insert into chart
+const count = [];
+
+// Fetch data from API asyncronously
+const getData = async function fetchDataFromURL() {
+  try {
+    const response = await fetch(url);
+    console.log("Request successful", response);
+
+    // Now turn data into a json readable format
+    data = await response.json();
+
+    // Initialize house & count values
+    getHouses(data);
+    getCount();
+
+    // Render chart with values
+    renderChart();
```

Comment:
The fact that these 3 functions are modifying/using global state makes it hard to understand the flow of data.

I would recommend returning arrays from the `getHouses` and `getCount` functions and then passing them explicitly as arguments to `renderChart`.

Reply:
This definitely makes it look a bit cleaner and easier to understand, thank you!